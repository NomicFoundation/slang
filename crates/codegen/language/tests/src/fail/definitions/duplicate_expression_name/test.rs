#![allow(unused_crate_dependencies)]

codegen_language_macros::compile!(Language(
    name = Foo,
    root_item = Bar,
    leading_trivia = Sequence([]),
    trailing_trivia = Sequence([]),
    versions = ["1.0.0"],
    sections = [Section(
        title = "Section One",
        topics = [Topic(
            title = "Topic One",
            items = [
                Precedence(
                    name = Bar,
                    precedence_expressions = [
                        PrecedenceExpression(name = Expression1, operators = []),
                        PrecedenceExpression(name = Expression2, operators = []),
                        PrecedenceExpression(name = Expression1, operators = [])
                    ],
                    primary_expressions = [PrimaryExpression(reference = Baz)]
                ),
                Token(
                    name = Baz,
                    definitions = [TokenDefinition(scanner = Atom("baz"))]
                )
            ]
        )]
    )]
));

fn main() {}
