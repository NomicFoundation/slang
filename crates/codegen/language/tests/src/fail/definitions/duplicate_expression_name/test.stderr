error: An expression with the name 'Expression1' already exists.
  --> src/fail/definitions/duplicate_expression_name/test.rs:20:53
   |
20 |                         PrecedenceExpression(name = Expression1, operators = [])
   |                                                     ^^^^^^^^^^^
